test/unit/range/views/view_minimiser_test.cpp

```diff
@@ -73,73 +108,142 @@ class minimiser_test : public ::testing::Test
 
     std::vector<seqan3::dna4> too_short_text{"AC"_dna4};
 
-    seqan3::bitcompressed_vector<seqan3::dna4> text3{"ACGGCGACGTTTAG"_dna4};
+    std::vector<seqan3::dna4> text3{"ACGGCGACGTTTAG"_dna4};
+    result_t result3_ungapped{26, 97, 27, 6, 1};  // ACGG, CGAC, ACGT, aacg, aaac - lowercase for reverse complement
+    result_t result3_gapped{2, 5, 3, 2, 1};       // A--G, C--C, A--T, a--g, a--c - "-" for gap
     result_t result3_ungapped_no_rev{26, 97, 27}; // ACGG, CGAC, ACGT
     result_t result3_gapped_no_rev{2, 5, 3};      // A--G, C--C-, A--T "-" for gap
-    result_t result3_ungapped_no_rev_stop{26, 97};// For stop at first T
-    result_t result3_gapped_no_rev_stop{2, 5};    // For stop at first T
+    result_t result3_ungapped_stop{26, 97};       // For stop at first T
+    result_t result3_gapped_stop{2, 5};           // For stop at first T
+    result_t result3_start{1};                    // For start at second A, ungapped and gapped the same
+    result_t result3_ungapped_no_rev_start{27};   // For start at second A
+    result_t result3_gapped_no_rev_start{3};      // For start at second A
 };
 
 TYPED_TEST(minimiser_view_properties_test, concepts)
 {
     TypeParam text{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4, 'C'_dna4, 'G'_dna4, 'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4,
                 'T'_dna4, 'T'_dna4, 'A'_dna4, 'G'_dna4}; // ACGTCGACGTTTAG
-    auto v = text | kmer_view | minimiser_view;
+
+    auto v = text | kmer_view | minimiser_no_rev_view;
     EXPECT_TRUE(std::ranges::input_range<decltype(v)>);
     EXPECT_TRUE(std::ranges::forward_range<decltype(v)>);
     EXPECT_FALSE(std::ranges::bidirectional_range<decltype(v)>);
     EXPECT_FALSE(std::ranges::random_access_range<decltype(v)>);
     EXPECT_TRUE(std::ranges::view<decltype(v)>);
     EXPECT_FALSE(std::ranges::sized_range<decltype(v)>);
     EXPECT_FALSE(std::ranges::common_range<decltype(v)>);
-    EXPECT_TRUE(seqan3::const_iterable_range<decltype(v)>);
+    EXPECT_EQ(seqan3::const_iterable_range<decltype((text | kmer_view))>,
+              seqan3::const_iterable_range<decltype(v)>);
     EXPECT_FALSE((std::ranges::output_range<decltype(v), size_t>));
+
+    if constexpr (std::ranges::bidirectional_range<TypeParam>) // excludes forward_list
+    {
+        auto v2 = text | kmer_view | seqan3::views::minimiser(5, text | rev_kmer_view);
+        EXPECT_TRUE(std::ranges::input_range<decltype(v2)>);
+        EXPECT_TRUE(std::ranges::forward_range<decltype(v2)>);
+        EXPECT_FALSE(std::ranges::bidirectional_range<decltype(v2)>);
```

Comment:
Why is this not preserved? Or why is this constrained for ranges that model bidirectional?
I mean sure it is not bidirectional, but why constraining this? What is with the tests above? I think this should be reused.

Reply:
Like the comment says, it is to exclude forward_lists from the test, because you can not reverse a forward list and that is what is done in the test.